frontend/src/modules/_shared/Figure.tsx

```diff
@@ -273,14 +440,20 @@ export function makeSubplots(options: MakeSubplotOptions): Figure {
                         anchor: anchorY,
                         domain: [yDomainStart, yDomainEnd],
                         showticklabels: true,
+                        tickangle: options.yAxisTickAngle,
                         showgrid: options.showGrid ?? false,
                     },
                 };
             }
 
-            if (options.subplotTitles && options.subplotTitles.length > index) {
-                const title = `<b>${options.subplotTitles[index]}</b>`;
-                annotations.push({
+            // If subplot titles are provided, add them as annotations.
+            // Set empty text if title is not provided for the subplot.
+            if (options.subplotTitles) {
+                const title =
+                    options.subplotTitles && options.subplotTitles !== true
```

Comment:
Another check if `options.subplotTitles` is defined does not seem necessary?

Also, should the titles maybe always be there and just set to "" when not given? This way, they might be updated later on in Figure.

Reply:
Changed from `subplotTitles?: true | string[]` to `subplotTitles?: string[]`, and always adding subplot title annotation. If `subplotTitles === undefined` or `subplotTitle[index]` is out of bounds, the text is set to empty string. Thereby annotations are always added when creating subplots.